vm/src/stdlib/socket.rs

```diff
@@ -372,6 +382,44 @@ fn get_addr_tuple(vm: &VirtualMachine, addr: SocketAddr) -> PyResult {
     Ok(vm.ctx.new_tuple(vec![ip, port]))
 }
 
+#[cfg(unix)]
+fn socket_gethostname(vm: &VirtualMachine) -> PyResult {
```

Comment:
You can use the same function as in windows. The `gothostname` module lib support it.

Reply:
Done :)